scripts/zones/Garlaige_Citadel/mobs/DE_Wanzo-Unzozo.lua

```diff
@@ -0,0 +1,240 @@
+        { -143.2720, 0.000000, 262.7013 },
+        { -138.2194, 0.000000, 230.2753 },
+        { -125.7615, 0.000000, 229.9893 },
+        { -126.1223, 0.000000, 217.9252 },
+        { -133.0517, 0.000000, 204.5436 }, -- Dead End
+        { -126.1223, 0.000000, 217.9252 },
+        { -125.7615, 0.000000, 229.9893 },
+        { -138.2194, 0.000000, 230.2753 },
+        { -138.9951, 0.000000, 254.4034 },
+        { -106.9740, 0.000000, 262.5682 },
+        { -100.8588, 0.000000, 256.1013 },
+        { -97.54940, 0.000000, 222.0876 },
+        { -65.87720, 0.000000, 217.7562 },
+        { -57.52250, 0.000000, 192.7869 },
+        { -60.73950, 5.927000, 153.6628 }, -- Basement
+        { -54.09130, 7.176200, 143.8619 },
+        { -42.03350, 6.000000, 144.9311 },
+        { -30.21860, 7.443100, 137.7087 },
+        { -17.81050, 6.567400, 147.5737 },
+        { -16.16890, 8.139700, 138.4585 },
+        { -20.54190, 6.000000, 118.7473 },
+        { -19.41390, 2.750000, 101.3711 },
+        { -101.4705, 0.000000, 100.9095 },
+        { -167.2571, 0.000000, 98.46240 },
+        { -177.5856, 0.000000, 94.07690 },
+        { -210.6050, 0.000000, 102.8821 },
+        { -220.0769, 0.000000, 110.8499 },
+        { -216.9912, 0.000000, 128.0300 },
+        { -219.6814, 0.000000, 139.8317 }, -- Destination
+
+    },
+    b =
+    {
+        { 0, 0, 0 },
+    },
+    c =
+    {
+        { 0, 0, 0 },
+    }
+}
+
+local path = paths.a -- Only one path works atm
+
+local escortProgress =
+{
+    NONE     = 0, -- Have not started pathing at all yet
+    ENROUTE  = 1, -- Started pathing and is continuing along path
+    PAUSED   = 2, -- Has paused pathing and awaiting orders
+    COMPLETE = 3, -- Has reached the end of the escort mission
+}
+
+local party = {}
+
+local entity = {}
+
+entity.shouldMove = function(mob, progress)
+    return not mob:isFollowingPath() and mob:getStatus() == xi.status.NORMAL and progress ~= escortProgress.COMPLETE
+end
+
+entity.onMobInitialize = function(mob)
+    mob:setLocalVar('progress', escortProgress.NONE)
+    mob:setSpeed(60)
+
+    mob:setMobMod(xi.mobMod.NO_MOVE, 1)
+    mob:setMobMod(xi.mobMod.NO_DESPAWN, 1)
+    mob:setAutoAttackEnabled(true)
+end
+
+entity.onMobRoam = function(mob)
+    mob:setStatus(xi.status.NORMAL)
+    local progress = mob:getLocalVar('progress')
+    local escortID = '[Escort]Wanzo'
+    local checkEscort = GetServerVariable(escortID)
+    local player = GetPlayerByID(mob:getLocalVar('player'))
+
+    local now = os.time()
+    local expire = mob:getLocalVar('expire')
+    if expire ~= 0 and expire <= now then
+        if progress ~= escortProgress.COMPLETE then
+            mob:showText(mob, ID.text.RAN_OUT_OF_TIME)
+        end
+
+        mob:setLocalVar('progress', escortProgress.NONE)
+        DespawnMob(mob:getID())
+        SetServerVariable(escortID, 0)
+        for _, v in ipairs(player:getParty()) do
+            utils.setQuestVar(v, 2, 88, 'Prog', 0)
```

Comment:
Should all these `utils.setQuestVar` calls be using enums instead of the raw area and quest IDs?

Reply:
I'd prefer to use `xi.quest.setVar` and deprecate `utils.setQuestVar` in this case, but also absolutely they should be using the proper enums